client/ayon_core/pipeline/farm/pyblish_functions.py

```diff
@@ -453,6 +475,54 @@ def prepare_representations(skeleton_data, exp_files, anatomy, aov_filter,
     return representations
 
 
+def _get_real_frames_to_render(frames):
+    """Returns list of frames that should be rendered.
+
+    Artists could want to selectively render only particular frames
+    """
+    frames_to_render = []
+    for frame in frames.split(","):
+        if "-" in frame:
+            splitted = frame.split("-")
+            frames_to_render.extend(
+                range(int(splitted[0]), int(splitted[1])+1))
+        else:
+            frames_to_render.append(int(frame))
+    frames_to_render.sort()
+    return frames_to_render
+
+
+def _get_real_files_to_rendered(collection, frames_to_render):
+    """Use expected files based on real frames_to_render.
+
+    Artists might explicitly set frames they want to render via Publisher UI.
+    This uses this value to filter out files
+    Args:
+        frames_to_render (list): of str '1001'
+    """
+    files = [os.path.basename(f) for f in list(collection)]
+    file_name, extracted_frame = list(collect_frames(files).items())[0]
+
+    if not extracted_frame:
+        return files
+
+    found_frame_pattern_length = len(extracted_frame)
+    normalized_frames_to_render = set()
+    for frame_to_render in frames_to_render:
+        normalized_frames_to_render.add(
+            str(frame_to_render).zfill(found_frame_pattern_length)
+        )
```

Comment:
Much faster than adding it one by one
```suggestion
    normalized_frames_to_render = {
        str(frame_to_render).zfill(found_frame_pattern_length)
        for frame_to_render in frames_to_render
    }
```